oauth2_http/java/com/google/auth/oauth2/ServiceAccountCredentials.java

```
@@ -312,11 +316,39 @@ public AccessToken refreshAccessToken() throws IOException {
     HttpRequest request = requestFactory.buildPostRequest(new GenericUrl(tokenServerUri), content);
     request.setParser(new JsonObjectParser(jsonFactory));
 
-    HttpResponse response;
-    try {
-      response = request.execute();
-    } catch (IOException e) {
-      throw new IOException("Error getting access token for service account: ", e);
+    ExponentialRetryAlgorithm retry =
+        new ExponentialRetryAlgorithm(
+            RetrySettings.newBuilder()
+                .setInitialRetryDelay(Duration.millis(100L))
+                .setRetryDelayMultiplier(1.3)
+                .setMaxRetryDelay(Duration.millis(60000L))
+                .setInitialRpcTimeout(Duration.millis(20000L))
+                .setRpcTimeoutMultiplier(1.0)
+                .setMaxRpcTimeout(Duration.millis(20000L))
+                .setTotalTimeout(Duration.millis(600000L))
+                .build());
+    TimedAttemptSettings attempt = retry.createFirstAttempt();
+
+    HttpResponse response = null;
+    int code = UNKNOWN_CODE;
+    while (retry.accept(attempt)) {
+      try {
+        try {
+          response = request.execute();
+          break;
+        } catch (SocketException | SocketTimeoutException exception) {
```

Nit: document that we are retrying here, not just swallowing the exceptions.